lib/errors.js

```diff
@@ -1,39 +1,73 @@
+'use strict';
+
+var util = require('util');
+
+/**
+ * The base class for all thinky errors
+ *
+ * @constructor
+ * @alias Error
+ */
+var ThinkyError = function(message) {
+  this.name = 'ThinkyError';
+  this.message = message || 'ThinkyError';
+  Error.captureStackTrace && Error.captureStackTrace(this, this.constructor);
+};
+util.inherits(ThinkyError, Error);
+
 /**
  * DocumentNotFound error which is returned when `get` returns `null`.
  * "Extends" Error
  */
-function DocumentNotFound(message) {
-  Error.captureStackTrace(this, DocumentNotFound);
+var DocumentNotFoundError = function(message) {
+  ThinkyError.call(this, message);
+  this.name = "DocumentNotFoundError";
```

Comment:
I moved it all to the constructor because I felt like it made the code more readable to move all manipulation to the same place. Definitely properties like e.g. message need to be computed on a case-by-case basis. Your argument is that it wastes resources or something to not define name as part of the prototype? I'm not sure it's warranted considering the number of potential error objects created in the lifetime of using this module, so I opt for readability in that case